channeldb/invoices.go

```diff
@@ -407,6 +410,32 @@ type InvoiceHTLC struct {
 	// CustomRecords contains the custom key/value pairs that accompanied
 	// the htlc.
 	CustomRecords record.CustomSet
+
+	// AMP is a copy of the AMP record presented in the onion payload
+	// containing the information necessary to correlate and settle a
+	// spontaneous HTLC set. Newly accepted legacy keysend payments will
+	// also have this field set as we automatically promote them into an AMP
+	// payment for internal processing.
+	//
+	// NOTE: This value will only be set for AMP HTLCs.
+	AMP *record.AMP
```

Comment:
currently the preimage is not set until all shards have arrived, so it would only be combining the `AMP` and `Hash` fields. However, there may be an argument to not persisting AMP HTLCs prematurely since someone could fill up your harddrive by sending probes that never settle. In that case we'd just track the sets in memory until everything has arrived, and then write everything all at once. I'm kinda leaning towards that approach now, which would mean they could be combined into one struct

Reply:
@halseth I went back and merged everything into a single struct, PTAL

Reply:
I like it 👍